src/material_library.cpp

```diff
@@ -0,0 +1,221 @@
+#include <unistd.h>
+#include <iostream>
+
+#ifndef PYNE_IS_AMALGAMATED
+#include "material_library.h"
+#endif
+
+namespace pyne {
+// Empty Constructor
+MaterialLibrary::MaterialLibrary(){};
+
+// Default constructor
+MaterialLibrary::MaterialLibrary(std::string file, std::string datapath) {
+  if (!check_file_exists(file)) {
+    std::cerr << "The file " << file << " does not exist or is read protected"
+              << std::endl;
+    exit(1);
+  }
+  if (!hdf5_path_exists(file, datapath)) {
+    std::cerr << "The datapath " << datapath << " in " << file << " empty."
+              << std::endl;
+    exit(1);
+  }
+
+  // load materials
+  from_hdf5(file);
+};
+
+// Destructor
+MaterialLibrary::~MaterialLibrary(){};
+
+// convert convert a filename into path+filename (for pyne)
+std::string MaterialLibrary::get_full_filepath(char* filename) {
+  std::string file(filename);
+  return MaterialLibrary::get_full_filepath(file);
+}
+
+// convert convert a filename into path+filename (for pyne)
+std::string MaterialLibrary::get_full_filepath(std::string filename) {
+  // remove all extra whitespace
+  filename.erase(std::remove(filename.begin(), filename.end(), ' '),
+                 filename.end());
+  // use stdlib call
+  const char* full_filepath = realpath(filename.c_str(), NULL);
+  return std::string(full_filepath);
+}
+
+// see if file exists
+bool MaterialLibrary::check_file_exists(std::string filename) {
+  // from http://stackoverflow.com/questions/12774207/
+  // fastest-way-to-check-if-a-file-exist-using-standard-c-c11-c
+  std::ifstream infile(filename.c_str());
```

Comment:
Nice! Haven't seen this check for a file before. If the file is successfully opened, we should close the `ifstream` before leaving this function for safety.

Reply:
I believe we are good here as the destructor closes the file by default.

"In case that an object is destroyed while still associated with an open file, the destructor automatically calls the member function close." from http://www.cplusplus.com/doc/tutorial/files/